src/tools/knowledge_graph_tools.py

```
@@ -0,0 +1,586 @@
+
+    except Exception as e:
+        return {"valid": False, "error": f"Failed to parse repository URL: {str(e)}"}
+
+
+async def check_ai_script_hallucinations(ctx: Context, script_path: str) -> str:
+    """
+    Analyzes a Python script for AI hallucinations by validating its structure against a Neo4j knowledge graph.
+
+    This function checks imports, method calls, class instantiations, function calls, and attribute accesses
+    in the given script, comparing them to real repository data in the knowledge graph. Returns a JSON string
+    with hallucination detection results, confidence scores, validation summaries, recommendations, and analysis metadata.
```

Ruff (0.11.9)

Line too long (122 > 120)

(E501)